File: problemas/1100/1117.dart
import 'dart:io';

void main() {
  double nota1 = double.parse(stdin.readLineSync());

  while (nota1 < 0 || nota1 > 10)
  {
    stdout.writeln('nota invalida');
    nota1 = double.parse(stdin.readLineSync());
  }

  double nota2 = double.parse(stdin.readLineSync());

  while (nota2 < 0 || nota2 > 10)
  {
    stdout.writeln('nota invalida');
    nota2 = double.parse(stdin.readLineSync());
  }

  stdout.writeln('media = ${((nota1 + nota2) / 2).toStringAsFixed(2)}');
}

File: problemas/1100/1117.py
nota1 = float(input())

while not(0 <= nota1 <= 10):
    print('nota invalida')
    nota1 = float(input())

nota2 = float(input())

while not(0 <= nota2 <= 10):
    print('nota invalida')
    nota2 = float(input())

print(f'media = {(nota1 + nota2) / 2:.2f}')
